modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,283 @@
+            .getPackageOf(classElement)
+            .getQualifiedName()
+            .toString()
+            .replace('.', '/');
+        String packagePrefix = !packageName.isEmpty() ? packageName + "/" : "";
+        String classHeader = packagePrefix + getFullClassName(classElement);
+        return classHeader.replace("$", "\\$") + "(\\$.+)?";
+    }
+
+    private String getFullClassName(Element classElement) {
+        List<String> parentClasses = new ArrayList<String>();
+        Element enclosingElement = classElement.getEnclosingElement();
+        while (enclosingElement != null &&
+            enclosingElement.getKind().isClass()) {
+            parentClasses.add(enclosingElement.getSimpleName().toString());
+            enclosingElement = enclosingElement.getEnclosingElement();
+        }
+        StringBuilder className = new StringBuilder();
+        for (int index = parentClasses.size() - 1; index >= 0; index--) {
+            className.append(parentClasses.get(index));
+            className.append("$");
+        }
+        return className.toString() + classElement.getSimpleName();
+    }
+
+    private String getMethodIdentifierString(Element methodElement) {
+        TypeVisitor<ExecutableType, Void> executableTypeVisitor =
+            new SimpleTypeVisitor6<ExecutableType, Void>() {
+                @Override
+                public ExecutableType visitExecutable(
+                    ExecutableType executableType, Void obj
+                ) {
+                    return executableType;
+                }
+            };
+        ExecutableType method =
+            methodElement.asType().accept(executableTypeVisitor, null);
+        List<? extends TypeMirror> methodParams = method.getParameterTypes();
+        TypeMirror returnType = method.getReturnType();
+        String returnTypeString = "";
+        if (returnType.getKind().equals(TypeKind.ARRAY)) {
+            returnTypeString = getArrayType(returnType);
+        } else {
+            returnTypeString = getParameterType(returnType);
+        }
+        String fullClassPattern =
+            getClassHeader(methodElement.getEnclosingElement());
+        String fullClassName =
+            fullClassPattern.substring(0, fullClassPattern.indexOf('('))
+            .replace("\\$", "$");
+        String methodArguments = "";
+        if (methodElement.getKind().equals(ElementKind.CONSTRUCTOR)) {
+            int index = fullClassName.lastIndexOf("$");
+            if (index != -1) {
+                methodArguments += fullClassName.substring(0, index) + " ";
+            }
+        }
+        methodArguments += getMethodArguments(methodParams);
+        methodArguments = methodArguments.replace('/', '.')
+            .replace("$", ".");
+        methodArguments = !methodArguments.isEmpty() ?
+            methodArguments.substring(0, methodArguments.length() - 1) :
+            methodArguments;
+        return fullClassName + " " + methodElement.getSimpleName() + " " +
```

Comment:
Is it worth it to make a helper method for constructing this? Something like `getMethodRep(String, String, String, List<String>)`. Then this could be shared between the processor and the plugin.